Extension method with default argument fails to construct

## Compiler version

v3.1.1-RC2

## Minimized code

https://scastie.scala-lang.org/MiSpvFT0SKygDNEJdezOzQ
```Scala
class Foo

extension (foo : Foo)
  def patch(arg : List[Int], arg2 : Int = 0) : Unit = {}
  def patch(arg : Int) : Unit = patch(List(arg))
```

## Output

```scala
value patch is not a member of Playground.Foo.
An extension method was tried, but could not be fully constructed:

    Playground.patch()
```

## Expectation
No error.

The problem comes from a combination of overloading, currying, and default arguments. It has nothing to do with extension methods. For instance:
```scala
class Foo

object Test {
  def patch2(foo: Foo)(arg : List[Int], arg2 : Int = 0) : Unit = {}
  def patch2(foo: Foo)(arg : Int) : Unit = patch2(foo)(List(arg))
}
```
fails with 
```
-- [E134] Type Error: i14278a.scala:5:43 ---------------------------------------
5 |  def patch2(foo: Foo)(arg : Int) : Unit = patch2(foo)(List(arg))
  |                                           ^^^^^^
  |None of the overloaded alternatives of method patch2 in object Test with types
  | (foo: Foo)(arg: Int): Unit
  | (foo: Foo)(arg: List[Int], arg2: Int): Unit
  |match arguments ((foo : Foo))
1 error found
```
Scala 2.13 also fails here:
```
14278a.scala:5: error: ambiguous reference to overloaded definition,
both method patch2 in object Test of type (foo: Foo)(arg: Int): Unit
and  method patch2 in object Test of type (foo: Foo)(arg: List[Int], arg2: Int): Unit
match argument types (Foo)
  def patch2(foo: Foo)(arg : Int) : Unit = patch2(foo)(List(arg))
                                           ^
1 error
```
Maybe the advice (requirement?) should be that overloaded curried methods (which includes extension methods) cannot have default arguments.

--- a/tests/pos/i14278.scala
+++ b/tests/pos/i14278.scala
@@ -1,0 +1,6 @@
+// https://github.com/lampepfl/dotty/issues/14278
+class Foo
+
+extension (foo: Foo)
+  def patch(arg: List[Int], arg2: Int = 0): Unit = {}
+  def patch(arg: Int): Unit = patch(List(arg))

